pkgs/io_file/lib/src/file_system.dart

```diff
@@ -30,6 +34,15 @@ class WriteMode {
 
 /// An abstract representation of a file system.
 base class FileSystem {
+  /// Create a directory at the given path.
+  ///
+  /// If the directory already exists, then `PathExistsException` is thrown.
+  ///
+  /// If the parent path does not exist, then `PathNotFoundException` is thrown.
+  void createDirectory(String path) {
+    throw UnsupportedError('createDirectory');
```

Comment:
Not necessarily for this PR, but if you create a utility method `unsupportedError(String)`, then this impl. could be:

```
void createDirectory(String path) => unsupportedError('createDirectory');
```

Reply:
I'm talking with @lrhn now about whether this class should just be `abstract`/`interface`. If it lands that way, then all of these will disappear.